classads/classads.go

```diff
@@ -62,6 +62,7 @@ func (c *ClassAd) String() string {
 		buffer.WriteString(" = ")
 		switch v := value.(type) {
 		case string:
+			v = strings.ReplaceAll(v, "\n", "")
```

Comment:
This doesn't make sense.

The next line removes all newlines from the string.  What does removing them here do?

Reply:
The new line converts all newlines into the new ClassAd syntax's escape for a newline, which _actually gets handled properly_ all the way through until it gets to the schedd, which is rejects a string containing a newline as invalid; see https://opensciencegrid.atlassian.net/browse/HTCONDOR-2379 for details.

Reply:
@Todd-L-Miller: To be clear - 

The contents of the message currently should contain two bytes, `\` and `n`, not the single byte `\n`.  So you're saying that these bytes are re-interpreted as the newline character and then the interpreted string is printed into the Hold message?

We could squash newlines specifically - but are there other forbidden characters?  What about \r or general escape sequences?  Should we be escaping `\` to `\\`?

Reply:
Talked a bit with Jaime about this over Slack.  He believes that `\n` is the only special character in old ClassAds (Todd: note that this disagrees with your comment in the JIRA ticket -- please let me know if you guys decide on a different conclusion; we're flexible!).

So here's what I propose:
1. Add a comment explaining why `\n` is special and needs to be escaped.  Just squashing it is too much "magic".
2. Replace `\n` with `\\n` before quoting the string.  This preserves what was originally there as best we can.
3. Find the error message generating this newline and remove the newline.  It obviously shouldn't be there in the first place!

Reply:
I removed the `\n` in the error thrown from the director so it should not show up anymore. However, I'm confused as to why to remove `\\n` instead of just `\n`. Because with the old director error message (with the `\n` at the end) if I replace `\n` it works and successfully removes it but if I replace `\\n` it does not work and the `\n` is still there in the error message still causing problems.

Reply:
I think the current discussion is leaning toward:

```suggestion
			v = strings.ReplaceAll(v, "\n", "\\n").ReplaceAll(v, "\r", "\\r")
```

for step (2).